src/loadgenerator/Dockerfile

```diff
@@ -12,16 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM python:3-slim as base
+FROM python:3.7.2-slim as base
 
 FROM base as builder
 
 RUN apt-get -qq update \
     && apt-get install -y --no-install-recommends \
-        g++
+        g++ libzmq3-dev
 
 COPY requirements.txt .
 
+RUN pip install --upgrade pip
```

Comment:
is this actually required? 
if you use a sufficiently high version of python:3.x you should be getting a good pip.

in fact this goes against "reproducible builds" mantra, because you're installing "whatever is the most recent pip available" in this command ––which is not a good idea for reproducibility.

Reply:
Done